### ISSUE TITLE
How to enable annotations?

### ISSUE THREAD

**Issue by keironlowe-edriving** [ScottBrenner/cfn-lint-action#273]
In the screenshot in the README it shows annotations with the issues highlighted. But when I've implemented the action the annotations aren't being added for issues.

CFN Lint is detecting the issue, just the annotations aren't added.

Thanks

**Comment by keironlowe-edriving** [comment]
For anyone else that comes across this, I resolved it by parsing the results and manually adding the annotations using the below workflow...

```
name: CI

on:
  pull_request:

jobs:
  cfn_lint:
    runs-on: ubuntu-latest

    outputs:
      cfn-results: ${{ steps.cfn_lint.outputs.cfn-results }}

    steps:
      - name: Checkout code
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Setup Python
        uses: actions/setup-python@v4
        with:
          python-version: '3.7'

      - name: Setup Cloud Formation Linter with Latest Version
        uses: scottbrenner/cfn-lint-action@v2

      - name: Print the Cloud Formation Linter Version & run Linter.
        id: cfn_lint
        run: |
          echo $(cfn-lint -t template.yaml -f json | jq '[.[] | { "file" : .Filename, "level": .Level | ascii_downcase, "line": .Location.End.LineNumber, "message": (.Rule.Id + " - " + .Message) }]') > /tmp/results.json
          echo 'cfn-results<<EOF' >> $GITHUB_OUTPUT
          cat /tmp/results.json >> $GITHUB_OUTPUT
          echo 'EOF' >> $GITHUB_OUTPUT      

  cfn_lint_annotations:
    runs-on: ubuntu-latest
    needs: [ cfn_lint ]
    if: needs.cfn_lint.outputs.cfn-results != '[]'

    strategy:
      matrix:
        value: ${{ fromJSON(needs.cfn_lint.outputs.cfn-results) }}

    steps:
      - name: Add Annotation
        run: echo "::${{ matrix.value.level }} file=${{ matrix.value.file }},line=${{ matrix.value.line }}::${{ matrix.value.message }}"
```

I'll run through some of the steps...

```
outputs:
  cfn-results: ${{ steps.cfn_lint.outputs.cfn-results }}
```
This defines that this job will have a certain output, and the value of that output is the output of the `cfn_lint` step.

```
- name: Print the Cloud Formation Linter Version & run Linter.
  id: cfn_lint
  run: |
    echo $(cfn-lint -t template.yaml -f json | jq '[.[] | { "file" : .Filename, "level": .Level | ascii_downcase, "line": .Location.End.LineNumber, "message": (.Rule.Id + " - " + .Message) }]') > /tmp/results.json
    echo 'cfn-results<<EOF' >> $GITHUB_OUTPUT
    cat /tmp/results.json >> $GITHUB_OUTPUT
    echo 'EOF' >> $GITHUB_OUTPUT 
```

This executes `cfn-lint` against the file with the output format as JSON. I'm then passing that JSON response to jq which filters it down into the exact format required. I'm then outputting the JSON into a local file. We're then echoing the content of that file as a multiline string as an output value for that step (which is then used as an output for the job as a whole)

```
  cfn_lint_annotations:
    runs-on: ubuntu-latest
    needs: [ cfn_lint ]
    if: needs.cfn_lint.outputs.cfn-results != '[]'

    strategy:
      matrix:
        value: ${{ fromJSON(needs.cfn_lint.outputs.cfn-results) }}

    steps:
      - name: Add Annotation
        run: echo "::${{ matrix.value.level }} file=${{ matrix.value.file }},line=${{ matrix.value.line }}::${{ matrix.value.message }}"
```

This sets up another job, and uses the JSON output of the `cfn_lint` job as a matrix. So it will execute this job for each item in the JSON array. We then just echo out a string in a [required format](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-creating-an-annotation-for-an-error) to create an annotation.

Hope that helps anybody else. Although ideally this would be handled by this action.

**Comment by ScottBrenner** [comment]
I wonder if the method of creating annotations has changed since introduced here in https://github.com/ScottBrenner/cfn-lint-action/commit/3e2d72a54e436a89d9fd555525335fedb8353e3e, leaving some links for reference

- https://github.com/actions/toolkit/tree/main/packages/core#annotations
- https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md
- https://github.com/actions/toolkit/blob/main/docs/commands.md#problem-matchers

**Comment by JonathanCrane-DOE** [comment]
@keironlowe-edriving as someone who came with the exact same issue, thank you so much for circling back and showing this.  I implemented and it works.
